src/codegen.cpp

```diff
@@ -1807,6 +1807,53 @@ static jl_cgval_t emit_invoke(jl_codectx_t &ctx, const jl_cgval_t &lival, const
 static Value *literal_pointer_val(jl_codectx_t &ctx, jl_value_t *p);
 static unsigned julia_alignment(jl_value_t *jt);
 
+static AttributeList get_attrs_ipoeffects(jl_codectx_t &ctx, std::pair<bool,uint32_t> effect, bool is_decl, bool has_ptr_arg)
+{
+    auto &context = ctx.builder.getContext();
+    if (!effect.first)
+        return AttributeList();
+    uint32_t effects = effect.second;
+    uint8_t consistent = (effects >> 0) & 0x07;
+    uint8_t effect_free = (effects >> 3) & 0x03;
+    bool nothrow = (effects >> 5) & 0x01;
+    bool terminates = (effects >> 6) & 0x01;
+    bool notaskstate = (effects >> 7) & 0x01;
+    uint8_t inaccessiblememonly = (effects >> 8) & 0x03;
+    bool nonoverlayed = (effects >> 10) & 0x01;
+    bool noinbounds = (effects >> 11) & 0x01;
+    AttrBuilder attr(context);
+
+    if (consistent == 0) {
+    }
+    if (effect_free == 0) {
+        // if (!has_ptr_arg)
+            // attr.addAttribute(Attribute::ReadOnly); //Not legal because of the ptls
+    }
+    if (nothrow == 1) {
+        attr.addAttribute(Attribute::NoUnwind);
+    }
+    if (terminates == 1){
+        attr.addAttribute(Attribute::WillReturn);
+    }
+    if (inaccessiblememonly == 0) {
+        attr.addAttribute(Attribute::InaccessibleMemOrArgMemOnly); // Can't be inaccesiblememonly because of the ptls
+    } else if (inaccessiblememonly == 2){
+        attr.addAttribute(Attribute::InaccessibleMemOrArgMemOnly);
+    }
+    if (notaskstate == 1) {
```

Comment:
Is that read only task state or task state at all?

Reply:
So, from the docs
```julia
## `:notaskstate`

The `:notaskstate` setting asserts that the method does not use or modify the
local task state (task local storage, RNG state, etc.) and may thus be safely
moved between tasks without observable results.
```